demo-java/app/src/main/res/layout/autocomplete_address_activity.xml

```diff
@@ -0,0 +1,158 @@
+        android:layout_height="wrap_content"
+        android:text="@string/autocomplete_address2_label"/>
+
+    <EditText
+        android:id="@+id/autocomplete_address2"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:autofillHints=""
+        android:imeOptions="actionNext"
+        android:inputType="text"/>
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/autocomplete_city_label"/>
+
+    <EditText
+        android:id="@+id/autocomplete_city"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:autofillHints=""
+        android:imeOptions="actionNext"
+        android:inputType="text"/>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal">
+
+      <LinearLayout
+          android:layout_width="0dp"
+          android:layout_height="wrap_content"
+          android:layout_weight="1"
+          android:orientation="vertical">
+
+      <TextView
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:text="@string/autocomplete_state_label"/>
+
+      <EditText
+          android:id="@+id/autocomplete_state"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:autofillHints=""
+          android:imeOptions="actionNext"
+          android:inputType="textCapCharacters"/>
+
+      </LinearLayout>
+
+      <LinearLayout
+          android:layout_width="0dp"
+          android:layout_height="wrap_content"
+          android:layout_weight="1"
+          android:orientation="vertical">
+
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/autocomplete_postal_label"/>
+
+      <EditText
+          android:id="@+id/autocomplete_postal"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:layout_weight="1"
+          android:autofillHints=""
+          android:imeOptions="actionNext"
+          android:inputType="number"/>
+      </LinearLayout>
+    </LinearLayout>
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/autocomplete_country_label"/>
+
+    <EditText
+        android:id="@+id/autocomplete_country"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:autofillHints=""
+        android:imeOptions="actionNext"
+        android:inputType="text"/>
+
+    <!-- The map for visual confirmation of the selected address -->
+    <!-- Stub to only load the map after Autocomplete prediction selection -->
+    <ViewStub
```

Comment:
Nice! More of our docs should teach techniques like this.

Reply:
One thing I'm unhappy with is that after he map inflates, I'm sending focus to Address 2 field (empty) so the soft keyboard is shown and the screen real estate is shrunk, and the "submit" button is hidden behidn the keyboard. Any ideas of how to suppress the soft keyboard but keep the focus cursor at Address2 so they can see the full form and submit button, and then only if they tap on Address2 the soft keyboard comes up? Otherwise they can just tap on Submit instead of having to scroll to find the Submit button.

Reply:
Maybe try one of the [windowSoftInputMode](https://developer.android.com/guide/topics/manifest/activity-element#wsoft) options, otherwise, you can [manually hide the keyboard](https://guides.codepath.com/android/Working-with-the-Soft-Keyboard#hiding-the-soft-keyboard-programmatically). Thanks for the snippet, @codepath!

Reply:
Opened Issue #180 to remember this fix.